typescript/sdk/src/deploy/core/HyperlaneCoreChecker.ts

```diff
@@ -168,4 +168,74 @@ export class HyperlaneCoreChecker<
       this.addViolation(violation);
     }
   }
+
+  async checkInterchainGasPaymaster(local: Chain): Promise<void> {
+    const coreContracts = this.app.getContracts(local);
+    const igp = coreContracts.interchainGasPaymaster.contract;
+
+    // Construct the violation, updating the actual & expected
+    // objects as violations are found.
+    // A single violation is used so that only a single `setGasOracles`
+    // call is generated to set multiple gas oracles.
+    const gasOraclesViolation: IgpGasOraclesViolation = {
+      type: CoreViolationType.InterchainGasPaymaster,
+      subType: IgpViolationType.GasOracles,
+      contract: igp,
+      chain: local,
+      actual: {},
+      expected: {},
+    };
+
+    // The `gasOracles` mapping was added in a new implementation of the IGP.
+    // If calling this reverts, we are still using an old implementation that
+    // must be upgraded, so we just consider the mapping as unset and default
+    // to address(0) for all gas oracles.
+    // Note that it's important that `checkProxiedContracts` is called before
+    // `checkInterchainGasPaymaster` - this way, the govern script will first
+    // change the IGP proxy to use the correct new implementation, and the
+    // IgpGasOraclesViolation will be able to be correctly handled.
```

Comment:
this seems fragile

Reply:
I think again we need this due to lack of fork testing

Reply:
agree this is fragile